things/service.go

```diff
@@ -495,31 +495,33 @@ func (ts *thingsService) Connect(ctx context.Context, token, chID string, thIDs
 		return err
 	}
 
-	if cgrID == "" {
-		return errors.ErrNotFound
-	}
+	if cgrID != "" {
+		var tgrIDs []string
+		for _, thID := range thIDs {
+			tgrID, err := ts.groups.RetrieveThingMembership(ctx, thID)
+			if err != nil {
+				return err
+			}
 
-	var tgrIDs []string
-	for _, thID := range thIDs {
-		tgrID, err := ts.groups.RetrieveThingMembership(ctx, thID)
-		if err != nil {
-			return err
+			if tgrID != "" {
```

Comment:
Check here if groupID is different. No nned to create an array for that. As sooner you catch the error as better it is.